bindings/postgres/postgres.go

```diff
@@ -212,6 +242,62 @@ func (p *Postgres) exec(ctx context.Context, sql string, args ...any) (result in
 	return res.RowsAffected(), nil
 }
 
+func (p *Postgres) fetchOne(ctx context.Context, sql string, args ...any) (result []byte, err error) {
+
+	rows, err := p.db.Query(ctx, sql, args...)
+	if err != nil {
+		return nil, fmt.Errorf("error executing query: %w", err)
+	}
+
+	rs := make(map[string]any, 0)
+	for rows.Next() {
```

Comment:
Since "fetchOne" is meant to fetch a single row, this method should return an error if the query returns more than one row.

Within the `for` loop, add a check to see if this is the 2nd iteration, and stop with an error if it is.